packages/idyll-compiler/src/lexer.js

```diff
@@ -392,7 +392,7 @@ const lex = function(options) {
     return ['EXPRESSION'].concat(formatToken(lexeme));
   });
 
-  lexer.addRule(/[+\-]?[0-9]+\.?[0-9]*/, function(lexeme) {
+  lexer.addRule(/[+\-]?\.?[0-9]+\.?[0-9]*/, function(lexeme) {
     this.reject = !inComponent;
```

Comment:
This looks like it would match `.123.456` and `.123.`. I think the easiest way to avoid this might be to set `this.reject = true` if there are two decimal points found in the number.